### ISSUE TITLE
Create generic plotting functions outside the `BolideDataFrame` class

### ISSUE THREAD

**Issue by jcsmithhere** [SETI/bolides#27]
There is a goal to utilize the functionality in this public bolides package as the low level plotting and visualization tools used within the bolide detection pipeline (which is closed source). An example is, the diagnostic tools within the pipeline plots detection density heat maps similarly to the `BolideDataFrame.plot_density`. We do not want duplicate code in two repositories, so the goal in this design change is to extract the main functionality of the `plot_density` method and place it in a helper module. This module will then be used by both `BolideDataFrame` and the detection pipeline.

**Comment by jcsmithhere** [comment]
There already is a `plotting.py` module. We can place the `plot_density` function in there. Other helper modules are called `*_utils.py`. There's a temptation to use that nomenclature, however the plotting utilities are going to be more full-feature functions, so I think we should continue to not use the `*_utils.py` naming convention.

**Comment by jcsmithhere** [comment]
The bolide detection pipeline density estimator uses `scipy.stats.gaussian_kde`. The bolides package uses `sklearn.neighbors.KernelDensity`. The latter seems more versatile because you can specify the kernel (not just Gaussian). Both are well respected and highly used packages so they probably both work very well.

**Comment by jcsmithhere** [comment]
Renaming issue to cover all plotting routines. No need to open a separate ticket. Also renamed branch.

**Comment by jcsmithhere** [comment]
One thing I noticed is the `plot_detections` method in `BolideDataFrame` uses `GeoDataFrame.to_crs` to change the coordinate frame for projecting on the map.  Once the plotting is finished the copy is deleted as we exit the method. `to_crs` makes a deepcopy of the whole `BolideDataFrame` object. This is quite memory inefficient. For very large dataframes this might be an issue. But, geopandas appears to be rather memory efficient, so maybe the chances of this being a problem is very low.

**Comment by jcsmithhere** [comment]
Two new funds created in `bolides.plotting.py` called `plot_scatter` and `plot_density`. These are now used by the BolideDataFrame methods `plot_detections` and `plot_desnity`. I believe I retained all functionality and this change should be transparent to any users.

There is also a thrid function created called `plotting.generate_plot`. This is a helper function to generate the figure and axis matplotlib objects but does not plot anything data. The returned fig and ax can then be passed to the plotting functions, wher einstead of generating a fig and ax, the plotting function just uses the one passed. It's used if one wishes to overlay different plots on the same figure.

**Comment by jcsmithhere** [comment]
This work appears to be complete. I am just waiting on the code review.

**Comment by jcsmithhere** [comment]
Still waiting for that code review...
